src/lib/native.ts

```diff
@@ -23,34 +24,37 @@ export interface AutoLaunchType {
   disable(): Promise<any>
 }
 
-export async function autoLaunch(): Promise<AutoLaunchType> {
-  const appName = process.env.DEV ? "app" : (await tauri.app.getName()).toString()
-  const os = (await tauri.os.type())
-  const appPath = await tauri.invoke('get_this_binary') as string
-  console.log('get_this_binary', appPath);
-  console.log(appPath);
-  let autoLauncher: osAL
-  if (os == 'Darwin') autoLauncher = macAL
-  else if (os == 'Windows_NT') autoLauncher = winAL
-  else autoLauncher = linAL
-  return <AutoLaunchType>{
-    async isEnabled(): Promise<boolean> {
-      const result = await autoLauncher.isEnabled(appName)
-      return result
-    },
-    async enable(): Promise<any> {
-      const child = await autoLauncher.enable({ appName, appPath, hidden: false })
-      return child
-    },
-    async disable(): Promise<any> {
-      const child = autoLauncher.disable(appName)
-      return child
-    }
+export class AutoLauncher {
+  protected autoLauncher: osAL = nullAL
+  appName: string = 'app'
+  appPath: string = ''
+  enabled: boolean = false
+  async isEnabled(): Promise<boolean> {
+    const result = await this.autoLauncher.isEnabled(this.appName)
+    this.enabled = result
+    return result
+  }
+  async enable(): Promise<any> {
```

Comment:
I don't currently need any return value but I may implement a return value in the future so I set it to any as a placeholder. I can remove it if you like.

Reply:
`any` should almost never be used. This is because it can be used in place of ANY type, breaking the whole idea of typing in TypeScript, making it useless since you break invariant in one place and the whole app can't be trusted anymore.

If you don't have a value to return, make it `Promise<void>` to express that explicitly. If you change your mind in the future you will have to update the signature and code that relied on it to be `Promise<void>` will stop compiling, which is what you want from a typed language.